cmd/cosign/cli/options/key.go

```diff
@@ -53,4 +56,13 @@ type KeyOpts struct {
 	// Modeled after InsecureSkipVerify in tls.Config, this disables
 	// verifying the SCT.
 	InsecureSkipFulcioVerify bool
+
+	// DefaultLoadOptions may be set to control the behaviour of
+	// `LoadDefaultSigner/Verifier` family of functions. Some public/private key
+	// types have ambiguities with regards to the signing algorithm to use (e.g.
+	// RSA can be RSASSA-PSS or RSASSA-PKCS1v15). This is a way to control that.
+	//
+	// By default, Ed25519ph is used for ed25519 keys and RSA-PKCS1v15 is used
+	// for RSA keys.
+	DefaultLoadOptions *[]signature.LoadOption
```

Comment:
I think I’m missing something, but where is this option set?

Reply:
In `sign_blob.go`:
```golang
	shouldUpload, err := ShouldUploadToTlog(ctx, ko, nil, tlogUpload)
	if err != nil {
		return nil, fmt.Errorf("upload to tlog: %w", err)
	}

	if !shouldUpload {
		// To maintain backwards compatibility with older cosign versions,
		// we do not use ed25519ph for ed25519 keys when the signatures are not
		// uploaded to the Tlog.
		ko.DefaultLoadOptions = &[]signature.LoadOption{}
	}

```